Sources/StructuredData.swift

```diff
@@ -1,20 +1,66 @@
 @_exported import C7
 
-public protocol StructuredDataInitializable {
+public enum StructuredDataError: ErrorProtocol {
+    case incompatibleType
+}
+
+public protocol StructuredDataFallibleInitializable: StructuredDataFallibleConstructable {
     init(structuredData: StructuredData) throws
 }
 
-public protocol StructuredDataRepresentable {
+extension StructuredDataFallibleInitializable {
+    public static func construct(with structuredData: StructuredData) throws -> Self {
+        return try self.init(structuredData: structuredData)
+    }
+}
+
+public protocol StructuredDataInitializable: StructuredDataFallibleInitializable, StructuredDataConstructable {
+    init(structuredData: StructuredData)
+}
+
+extension StructuredDataInitializable {
+    public init(structuredData: StructuredData) throws {
+        self.init(structuredData: structuredData)
+    }
+
+    public static func construct(with structuredData: StructuredData) -> Self {
+        return self.init(structuredData: structuredData)
+    }
+}
+
+public protocol StructuredDataFallibleRepresentable {
+    func asStructuredData() throws -> StructuredData
+}
+
+public protocol StructuredDataRepresentable: StructuredDataFallibleRepresentable {
     var structuredData: StructuredData { get }
 }
 
-public protocol StructuredDataConvertible: StructuredDataInitializable, StructuredDataRepresentable {}
+extension StructuredDataRepresentable {
+    public func asStructuredData() throws -> StructuredData {
```

Comment:
maybe this protocol won't be needed at all.. we're still thinking about this. `make`  could be a better name. But I'm not sure.. @bradhilton what do you think?